utils/protobuf-schema-utilities/src/main/java/io/apicurio/registry/utils/protobuf/schema/FileDescriptorUtils.java

```diff
@@ -463,7 +463,8 @@ private static DescriptorProto messageElementToDescriptorProto(
                     allNestedTypes.put(field.getLocation(), mapMessage.build());
                 }
 
-                String jsonName = field.getDeclaredJsonName();
+                String jsonName = getDefaultJsonName(field.getName()).equals(field.getDeclaredJsonName())
+                        ? null : field.getDeclaredJsonName();
```

Comment:
Is this applicable for map fields?

Reply:
Yes it is applicable for both Map itself as well as Map key/value fields. Map itself is tested. Map key/value fields are auto generated and can also be taken care of by the logic.